include/aos/common/tools/array.hpp

```diff
@@ -642,6 +642,19 @@ class StaticArray : public Array<T> {
         Array<T>::operator=(array);
     }
 
+    /**
+     * Assigns static array from another  array.
+     *
+     * @param array array to assign from.
+     */
+    StaticArray& operator=(const Array<T>& array)
+    {
+        Array<T>::SetBuffer(mBuffer);
```

Comment:
It seems like buffer is not supposed to be changed since creation. So setBuffer should be called in constructors only.